mobile/Ihc/services/LocalDataService.js

```diff
@@ -224,6 +225,87 @@ export function getTriage(patientKey, strDate) {
   return triage;
 }
 
+// Create LabRequest
+export function enqueueLabRequest(labRequest) {
+  const timestamp = new Date().getTime();
+  // just incase object passed in does not already have key initialized
+  labRequest.key = LabRequest.makeKey(labRequest);
+  const existingLabRequest = realm.objects('LabRequest')
+    .filtered('key = "' + labRequest.key + '"')['0'];
+
+  labRequest.lastUpdate = timestamp;
+
+  if (existingLabRequest) {
+    throw new Error('LabRequest already exists');
+  }
+
+  realm.write( () => {
+    realm.create('LabRequest', labRequest);
+  });
+}
+
+// Delete LabRequest
+export function dequeueLabRequest(key) {
+  const labRequest = realm.objects('LabRequest')
+    .filtered('key = "' + key + '"')['0'];
+
+  if (!labRequest) {
+    throw new Error('LabRequest does not exist');
+  }
+
+  realm.write( () => {
+    realm.delete(labRequest);
+  });
+}
+
+// Update a LabRequest
+export function updateLabRequest(key, update) {
+  const existingLabRequest = realm.objects('LabRequest')
+    .filtered('key = "' + key + '"')['0'];
+
+  if (!existingLabRequest) {
+    throw new Error('LabRequest does not exist');
+  }
+
+  realm.write( () => {
+    const properties = Object.keys(LabRequest.schema.properties);
+    properties.forEach( p => {
+      if (p !== 'patientKey' && p !== 'key') {
```

Comment:
None of the fields that make up the key should be changed, aka:
````${labRequest.patientKey}&${labRequest.testType}${labRequest.dateRequested}````
You might want to put some comment or test accounting for this, so if someone changes the key formula, this doesn't get messed up